lib/wallaby/browser.ex

```diff
@@ -527,13 +527,35 @@ defmodule Wallaby.Browser do
 
   @doc """
   Executes javascript synchoronously, taking as arguments the script to execute,
-  and optionally a list of arguments available in the script via `arguments`
+  an optional list of arguments available in the script via `arguments`, and an
+  optional callback function with the result of script execution as a parameter.
   """
+  @spec execute_script(parent, String.t) :: parent
   @spec execute_script(parent, String.t, list) :: parent
+  @spec execute_script(parent, String.t, ((binary()) -> any())) :: parent
+  @spec execute_script(parent, String.t, list, ((binary()) -> any())) :: parent
 
-  def execute_script(session, script, arguments \\ []) do
+
+  def execute_script(session, script) do
+    {:ok, _} = Driver.execute_script(session, script, [])
+    session
+  end
+
+  def execute_script(session, script, arguments) when is_list(arguments) do
+    {:ok, _} = Driver.execute_script(session, script, arguments)
+    session
+  end
+
+  def execute_script(session, script, callback) when is_function(callback) do
+    {:ok, value} = Driver.execute_script(session, script, [])
+    callback.(value)
+    session
+  end
```

Comment:
Hm not sure, this seems like we are implementing the same function 4 times (3 here + the one below). I'm wondering if we could do anything through the use of default arguments. E.g. arguments defaults to `[]` and function defaults to `nil`.

I'd be fine with the callback function only being available if you pass in the arguments explicitly (even if it's `[]`). That should work then.

Alternatively the more abstracted functions (without args or callback functions) could call the more detailed functions to avoid the duplication.

Wdyt @keathley ? :)